### PR TITLE
refactor(curriculum): update lessons for arrow function music player project

### PR DESCRIPTION
**Pull request by jdwilkin4**
## Summary of changes
A lot of campers on the forum and discord are confused on how arrow functions work. Since arrow function play a major role in this certification, I have added lessons to break it down a little bit better. 

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

### CODE REVIEW

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/660aef1d5bd6c092f1931830.md [hunk 1]
```diff
@@ -0,0 +1,662 @@
+---
+id: 660aef1d5bd6c092f1931830
+title: Step 14
+challengeType: 0
+dashedName: step-14
+---
+
+# --description--
+
+Just regular functions, arrow functions can return values.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/660aef1d5bd6c092f1931830.md:10]:
```suggestion
Just like regular functions, arrow functions can return values. 
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/660af2b86c0fca95f7570fc5.md [hunk 2]
```diff
@@ -0,0 +1,642 @@
...
+
+```js
+const getTax = (price) => {
+  const taxRate = 0.08;
+  const tax = price * taxRate;
+  return tax;
+};
+```
+
+Refactor(update) your `addTwoNumbers` function to remove the `return` keyword and the curly braces `{}`. Your `addTwoNumbers` function should instead use an implicit return.
```
Comment (naomi-lgbt) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/660af2b86c0fca95f7570fc5.md:26]:
```suggestion
Refactor, or update, your `addTwoNumbers` function to remove the `return` keyword and the curly braces `{}`. Your `addTwoNumbers` function should instead use an implicit return.
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]
Just a couple of nits.

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/660aef1d5bd6c092f1931830.md (hunk 1 above)
Comment (jdwilkin4) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/660aef1d5bd6c092f1931830.md:10]: ohh.. hahaha 😆
good catch

english is hard 🤣

Comment (naomi-lgbt) [reply curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/660aef1d5bd6c092f1931830.md]: Harder than JavaScript, tbh.

**Review by naomi-lgbt — APPROVED** [review]

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/660ae3eeef9ad289bece426b.md [hunk 3]
```diff
@@ -0,0 +1,640 @@
...
+assert.isFunction(printGreeting);
+```
+
+You should use arrow syntax to create the `printGreeting` function.
+
+```js
+assert.match(code, /(let|const)\s+printGreeting\s*=\s*\(.*\)\s*=>\s*{\s*[\s\S]*}\s*;?/)
+```
+
+Your `printGreeting` function should include a `console.log('Hello there!)`.
```
Comment (Ksound22) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/660ae3eeef9ad289bece426b.md:44]:
```suggestion
Your `printGreeting` function should include a `console.log('Hello there!')`.
```

File: curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/660ae3eeef9ad289bece426b.md [hunk 4]
```diff
@@ -0,0 +1,640 @@
...
+
+```js
+assert.match(printGreeting.toString(), /console\.log\(['"]Hello\s+there!['"]\)/);
```
Comment (Ksound22) [inline curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-basic-string-and-array-methods-by-building-a-music-player/660ae3eeef9ad289bece426b.md:47]: Can we also allow backticks as that's also valid? I as a learner can go crazy and decided to test that out.

```suggestion
assert.match(printGreeting.toString(), /console\.log\(['"`]Hello\s+there!['"`]\)/);
```

**Comment by Ksound22** [comment]
I spotted one omission. Other than that, I'll be happy to approve.

**Review by Ksound22 — COMMENTED** [review]
LGTM

**Review by Ksound22 — APPROVED** [review]
LGTM 🚀

**Pull request merged**